File: codeforces/1475B New Year's Number.cpp
#include <iostream>
using namespace std;

#define ll long long

int main() {

	int t;
	cin >> t;
	while (t--) {
		int n, count = 1;
		cin >> n;
		while (n > 2020)
			n -= 2020, count++;
		if ((n > 0 && n < count) || n == 2020 || n == 2021)
			cout << "YES" << endl;
		else
			cout << "NO" << endl;
	}
	
}

File: codeforces/1475B New Year's Number.py
t = int(input())
while t > 0:
	t -= 1
	n = int(input())
	count = 1
	while n > 2020:
		n -= 2020
		count += 1
	if n in range(0, count) or n == 2020 or n == 2021:
		print('YES')
	else:
		print('NO')
